dashboard/app/controllers/courses_controller.rb

```diff
@@ -82,99 +87,88 @@ def new
   end
 
   def create
-    unit_group = UnitGroup.new(
+    @unit_group = UnitGroup.new(
       name: params.require(:course).require(:name),
       family_name: params.require(:family_name),
       version_year: params.require(:version_year),
       has_numbered_units: true
     )
-    if unit_group.save
-      unit_group.write_serialization
-      redirect_to action: :edit, course_name: unit_group.name
+    if @unit_group.save
+      @unit_group.write_serialization
+      redirect_to action: :edit, course_name: @unit_group.name
     else
-      render 'new', locals: {unit_group: unit_group}
+      render 'new', locals: {unit_group: @unit_group}
     end
   end
 
   def update
-    unit_group = UnitGroup.find_by_name!(params[:course_name])
-    unit_group.persist_strings_and_units_changes(params[:scripts], params[:alternate_units], i18n_params)
-    unit_group.update(course_params)
-    unit_group.write_serialization
-    CourseOffering.add_course_offering(unit_group)
-    unit_group.reload
-
-    unit_group.update_teacher_resources(params[:resourceTypes], params[:resourceLinks]) unless unit_group.has_migrated_unit?
-    if unit_group.has_migrated_unit? && unit_group.course_version
-      unit_group.resources = params[:resourceIds].map {|id| Resource.find(id)} if params.key?(:resourceIds)
-      unit_group.student_resources = params[:studentResourceIds].map {|id| Resource.find(id)} if params.key?(:studentResourceIds)
+    @unit_group.persist_strings_and_units_changes(params[:scripts], params[:alternate_units], i18n_params)
+    @unit_group.update(course_params)
+    @unit_group.write_serialization
+    CourseOffering.add_course_offering(@unit_group)
+    @unit_group.reload
+
+    @unit_group.update_teacher_resources(params[:resourceTypes], params[:resourceLinks]) unless @unit_group.has_migrated_unit?
+    if @unit_group.has_migrated_unit? && @unit_group.course_version
+      @unit_group.resources = params[:resourceIds].map {|id| Resource.find(id)} if params.key?(:resourceIds)
+      @unit_group.student_resources = params[:studentResourceIds].map {|id| Resource.find(id)} if params.key?(:studentResourceIds)
     end
 
-    unit_group.reload
-    render json: unit_group.summarize
+    @unit_group.reload
+    render json: @unit_group.summarize
```

Comment:
❤️  that this PR lets us set `@unit_group` in a consistent way across most controller actions. thanks for adding that part of this!